### ISSUE TITLE
Move zowe-cli to GitHub Projects

### ISSUE THREAD

**Issue by adam-wolfe** [zowe/zowe-cli#1990]
**Is your feature or enhancement request related to a problem or limitation? Please describe**

During the 12/14/2023 TSC call, it was suggested that squads adopt GitHub Projects as an alternative to ZenHub.

**Describe your enhancement idea**

Create a GitHub Projects project for this repository, implement a board and workflows to move issues and PRs to the correct status (pipeline) based on applied tags and automatically move new issues to a 'New Issues' status.

**Describe alternatives you've considered**

**Provide any additional context**

**Comment by t1m0thyj** [comment]
Here is a GitHub Projects board we can try out: https://github.com/orgs/zowe/projects/21

I wrote a script to import all our issues and PRs into the board, which I plan to clean up and make available next week.

I also encountered a few limitations:
1. With the built-in rules, all new issues (including pull requests) must be added to the same column. So new PRs will start in the "New Issues" column, instead of "In Progress" where we want them to be.
1. Issues don't automatically move into the Priority columns when a priority label is assigned. However this isn't any worse than ZenHub which had the same limitation.
1. Maximum number of issues per project is 1200 so we can't include all our closed issues since the beginning of time. I excluded from the import any issues closed more than 90 days ago.
1. Issue links open in a side panel like ZenHub, but pull request links open in a new tab which is a bit annoying IMO. There is a GitHub enhancement request related to this: https://github.com/orgs/community/discussions/59684

I think 1-3 could all be solved with GitHub workflows. We could create a workflow in each repo that uses [this action](https://github.com/marketplace/actions/move-issues-to-column) to move issues across columns whenever an issue is labeled or a PR is created, although this seems like a lot of effort to maintain in each repo. Alternatively, we could create a workflow in a single repo that runs daily and updates the entire board.
